solar_system.rb

```diff
@@ -0,0 +1,262 @@
+# Create a Planet class which will represent a planet.
+# Add an initialize method which takes several arguments and uses them to set the class' instance variables.
+# Create a method that returns the Planet's attributes in an easy to read fashion.
+# Create reader methods to give a user access to read the instance variables.
+# Make your SolarSystem class take an array of Planets instead of hashes.
+
+# Verify user input for integers
+def verify(user_input)
+  until user_input.to_i.to_s == user_input
+    puts "Please enter a number: "
+    user_input = gets.chomp
+  end
+  return user_input.to_i
+end
+
+# Verify user input for "Y" or "N"
+def verify_y_n(user_input)
+  until user_input == "Y" || user_input == "N"
+    puts "Please enter Y or N: "
+    user_input = gets.chomp.upcase
+  end
+  return user_input
+end
+
+class Planets
+
+  attr_reader :name, :color, :shape, :motto, :planetary_bird, :year_length, :distance_from_the_sun
+
+  def initialize(name, color, shape, motto, planetary_bird, year_length, distance_from_the_sun)
+
+    @name = name
+    @color = color
+    @shape = shape
+    @motto = motto
+    @planetary_bird = planetary_bird
+    @year_length = year_length
+    @distance_from_the_sun = distance_from_the_sun
+
+  end
+
+  def all_attributes
+    list = ""
+    list << "Name: #{@name}\nColor: #{@color}\nShape: #{@shape}\nMotto: #{@motto}\nPlantary bird: #{@planetary_bird}\nYear length: #{@year_length}\nDistance from the sun: #{@distance_from_the_sun}"
+    return list
+  end
+
+end
+# Create a SolarSystem class with an @planets instance variable
+
+class SolarSystem
+
+  # Create an initialize method which should take a collection of planet names
+  # and store them in an @planets instance variable
+  # Instead of Strings for planets, modify SolarSystem's initialize method to take
+  # a list of hashes where each planet is sent as a hash with at least 5 attributes.
+  def initialize(planets)
+    @planets = planets
+    @age = 1000  # Solar system age (in earth years) => 365,000 days
+  end
+
+  # Method to return just the planet names
+  def list_planet_names
+    planet_names = ""
+
+    @planets.each_with_index do |planet, i|
+      planet_names << "#{i+1}: #{planet.name}\n"
+    end
+
+    # Add Exit option
+    planet_names << "#{@planets.length + 1}: Exit"
+
+    return planet_names
+
+  end
+
+  # Create a method to add a planet to the list
+
+  def add_planet(new_planet)
+    @planets << new_planet
+  end
+
+  # Create a method which will return not print a list of the planets as a String
+  def return_planets
+    list = ""
+    @planets.each_with_index do |planet, i|
+      list << "--- #{i+1} ---\nName: #{planet.name}\nColor: #{planet.color}\nShape: #{planet.shape}\nMotto: #{planet.motto}\nPlanetary bird: #{planet.planetary_bird}\nYear length (in days): #{planet.year_length}\nDistance from the sun (in feet): #{planet.distance_from_the_sun}\n"
+    end
```

Comment:
This seems very similar to the code you wrote in `Planet#all_attributes` - would it be possible to consolidate the two, maybe by calling `all_attributes` here rather than formatting the string from scratch?